projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts

```diff
@@ -312,6 +312,116 @@ describe('IgxTreeGrid - Expanding / Collapsing', () => {
             verifyGridPager(fix, 4, '147', '1 of 1', [true, true, true, true]);
             expect(treeGrid.totalPages).toBe(1);
         });
+
+        it('Should update the paginator when a row of any level is expanded', () => {
+            // Test prerequisites
+            treeGrid.paging = true;
+            treeGrid.perPage = 5;
+            fix.detectChanges();
+            treeGrid.collapseAll();
+            fix.detectChanges();
+
+            // Verify current page
+            verifyGridPager(fix, 4, '147', '1 of 1', [true, true, true, true]);
+            expect(treeGrid.totalPages).toBe(1);
+
+            // Expand a row
+            const rowsDOM = TreeGridFunctions.getAllRows(fix);
+            let indicatorDivDOM = TreeGridFunctions.getExpansionIndicatorDiv(rowsDOM[3]);
+            indicatorDivDOM.triggerEventHandler('click', new Event('click'));
+            fix.detectChanges();
```

Comment:
Just a note, triggerEventHandler does not need to be followed by a fixture.detectChanges().